babel/src/test/java/org/apache/calcite/test/BabelParserTest.java

```diff
@@ -1153,6 +1153,27 @@ private void checkParseInfixCast(String sqlType) {
     sql(sql).ok(expected);
   }
 
+  @Test public void testCreateFunctionSqlFormNotDeterministicReturnsNullHasSecurityDefiner() {
+    final String sql =
+        "create function foo() "
+            + "returns Integer "
+            + "language sql "
+            + "returns null on null input "
+            + "not deterministic "
```

Comment:
This controversial test case is added. Where "returns null on null input" and "not deterministic" came in in one order but unparsed into another order. It would be grammatically the same query, since the order of "[not] deterministic", "[return null|called] on null input", and "specific <function_name>" is arbitrary. After all, it did unparse into a different query with equivalent meaning, should we be concerned about this behavior? and if so, any suggestions on how to unparse in the order of query input?

Reply:
No, as long as the behavior is the same, the order shouldn't matter.

Reply:
Okay. Thanks for clarifying.